src/transformers/tokenization_utils_fast.py

```diff
@@ -688,7 +688,19 @@ def train_new_from_iterator(
                 if getattr(self, f"_{token}") is not None:
                     special_token = getattr(self, token)
                     if special_token in special_tokens_map:
-                        kwargs[token] = special_tokens_map[special_token]
+                        special_token_full = getattr(self, f"_{token}")
+                        if isinstance(special_token_full, AddedToken):
+                            # Create an added token with the same paramters except the content
+                            signature = dict(inspect.signature(AddedToken).parameters)
+                            signature.pop("self", None)
+                            kwargs_added_token = {}
+                            for arg in signature.keys():
+                                kwargs_added_token[arg] = getattr(special_token_full, arg)
+                            kwargs_added_token["content"] = special_tokens_map[special_token]
+                            new_added_token = AddedToken(**kwargs_added_token)
```

Comment:
This is a bit too complicated for a class where the signature is fixed. We can just do:
```
kwargs[token] = AddedToken(
    special_tokens_map[special_token],
    single_word=special_token_full.single_word,
    lstrip=special_token_full.lstrip,
    rstrip=special_token_full.rstrip,
    normalized=special_token_full.normalized,
)
```

Reply:
Indeed, there was no need to make it so complicated. I apply your suggestion :slightly_smiling_face: